executor/aggregate_test.go

```diff
@@ -1168,3 +1171,110 @@ func (s *testSuiteAgg) TestIssue19426(c *C) {
 	tk.MustQuery("select a, b, sum(case when a < 1000 then b else 0.0 end) over (order by a) from t").
 		Check(testkit.Rows("1 11 11.0", "2 22 33.0", "3 33 66.0", "4 44 110.0"))
 }
+
+func reconstructParallelGroupConcatResult(rows [][]interface{}) []string {
+	data := make([]string, 0, len(rows))
+	for _, row := range rows {
+		if str, ok := row[0].(string); ok {
+			tokens := strings.Split(str, ",")
+			sort.Slice(tokens, func(i, j int) bool {
+				return tokens[i] < tokens[j]
+			})
+			data = append(data, strings.Join(tokens, ","))
+		}
+	}
+
+	sort.Slice(data, func(i, j int) bool {
+		return data[i] < data[j]
+	})
+
+	return data
+}
+
+func (s *testSuiteAgg) TestParallelStreamAggGroupConcat(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("use test;")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("CREATE TABLE t(a bigint, b bigint);")
+
+	for i := 0; i < 10000; i++ {
+		tk.MustExec("insert into t values(?, ?);", rand.Intn(100), rand.Intn(100))
+	}
+
+	sql := "select /*+ stream_agg() */ group_concat(a, b) from t group by b;"
+	concurrencies := []int{1, 2, 4, 8}
+	var expected []string
+	for _, con := range concurrencies {
+		tk.MustExec(fmt.Sprintf("set @@tidb_streamagg_concurrency=%d", con))
+		if con == 1 {
+			expected = reconstructParallelGroupConcatResult(tk.MustQuery(sql).Rows())
```

Comment:
How about adding a sort in the SQL so that we can remove `reconstructParallelGroupConcatResult`?
`select /*+ stream_agg() */ group_concat(a, b) from t group by b;` -> `select /*+ stream_agg() */ group_concat(a, b order by a, b) from t group by b order by b;`

Reply:
this mighe be useful, I will check this.

Reply:
no, this is not working at now. if we add sort, the current implementation will not attach shuffle, this can be fixed in the future version.